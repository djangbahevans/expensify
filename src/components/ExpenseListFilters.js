import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';
// import 'react-dates/initialize';

export class ExpenseListFilters extends React.Component {
    state = {
        calenderFocused: null
    };

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = (calenderFocused) => {
        this.setState(() => ({
            calenderFocused
        }));
    };

    onSortChange = e => {
        switch (e.target.value) {
            case 'date':
                this.props.sortByDate()
                break;
            case 'amount':
                this.props.sortByAmount();
                break;
        }
    };

    onTextChange = e => {
        this.props.setTextFilter(e.target.value);
    };

    render() {
        return (
            <div>
            <input
                type="text" value={this.props.filters.text}
                onChange={this.onTextChange} />
            <select
                value={this.props.filters.sortBy}
                onChange={this.onSortChange}>
                <option value="date">Date</option>
                <option value="amount">Amount</option>
            </select>
            <DateRangePicker
                startDate={this.props.filters.startDate}
                startDateId='startDate'
                endDate={this.props.filters.endDate}
                endDateId='endDate'
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calenderFocused}
                onFocusChange={this.onFocusChange}
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={() => false} />
        </div>
    )}
};

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

const mapDispatchToProps = dispatch => ({
    setTextFilter: text => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: startDate => dispatch(setStartDate(startDate)),
    setEndDate: endDate => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);