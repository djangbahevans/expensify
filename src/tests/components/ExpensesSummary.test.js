import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render correctly ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={15630} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render correctly ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={12} expensesTotal={15630} />);
    expect(wrapper).toMatchSnapshot();
});