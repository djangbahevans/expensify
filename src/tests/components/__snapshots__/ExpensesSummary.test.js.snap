// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expense
    <Component />
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render correctly ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    12
     expense
    s
     totalling 
    $0.00
  </h1>
</div>
`;
